scalameter-core/src/main/scala/org/scalameter/Parameter.scala

```diff
@@ -0,0 +1,86 @@
+    override final def skipWhitespace = false
+
+    private lazy val value: PackratParser[AnyValManifest[_]] = {
+      ("Byte" ^^^ Manifest.Byte) |
+        ("Short" ^^^ Manifest.Short) |
+        ("Char" ^^^ Manifest.Char) |
+        ("Int" ^^^ Manifest.Int) |
+        ("Long" ^^^ Manifest.Long) |
+        ("Float" ^^^ Manifest.Float) |
+        ("Double" ^^^ Manifest.Double) |
+        ("Boolean" ^^^ Manifest.Boolean) |
+        ("Unit" ^^^ Manifest.Unit)
+    }
+
+    private lazy val phantom: PackratParser[Manifest[_]] = {
+      ("Any" ^^^ Manifest.Any) |
+        ("Object" ^^^ Manifest.Object) |
+        ("AnyRef" ^^^ Manifest.AnyRef) |
+        ("AnyVal" ^^^ Manifest.AnyVal) |
+        ("Null" ^^^ Manifest.Null) |
+        ("Nothing" ^^^ Manifest.Nothing)
+    }
+
+    private def cls: PackratParser[String] = rep1sep(ident, ".") ^^ { case s =>
+      s.mkString(".")
+    }
+
+    private lazy val singleton: PackratParser[Manifest[_]] = cls ^? { case s if s.endsWith(".type") =>
+      val clazz = Class.forName(s.stripSuffix(".type")+"$")
+      Manifest.singleType(clazz.getField("MODULE$").get(null))
+    }
+
+    private lazy val classType: PackratParser[Manifest[_]] = opt(manifest <~ "#") ~ cls ~ opt("[" ~> rep1sep(manifest, ", ") <~ "]") ^^ {
+      case None ~ s ~ None => Manifest.classType(Class.forName(s))
+      case None ~ s ~ Some(targ :: targs) => Manifest.classType(Class.forName(s), targ, targs: _*)
+      case Some(p) ~ s ~ targs => Manifest.classType(p, Class.forName(s), targs.getOrElse(Nil): _*)
+    }
+
+    private lazy val array: PackratParser[Manifest[_]] = "Array" ~ "[" ~> manifest <~ "]" ^^ { case m =>
+      Manifest.arrayType(m)
+    }
+
+    private lazy val wildcard: PackratParser[Manifest[_]] = "_" ~> (opt(" >: " ~> manifest) ~ opt(" <: " ~> manifest)) ^^ {
+      case None ~ None => Manifest.wildcardType(Manifest.Nothing, Manifest.Any)
+      case None ~ Some(m) => Manifest.wildcardType(Manifest.Nothing, m)
+      case Some(m) ~ None => Manifest.wildcardType(m, Manifest.Any)
+      case Some(m1) ~ Some(m2) => Manifest.wildcardType(m1, m2)
+    }
+
+    private lazy val intersectionType: PackratParser[Manifest[_]] = rep1sep(manifest, " with ") ^^ { case ms =>
+      Manifest.intersectionType(ms: _*)
+    }
+
+    private[Parameter] lazy val manifest: PackratParser[Manifest[_]] =
+      value | phantom | singleton | array | wildcard | classType | intersectionType
+  }
+
+  def fromString(str: String): Parameter[_] = {
+    val i = str.lastIndexOf('|')
+    if (i == -1) sys.error("""Invalid parameter string. It should have following form "name|manifest".""")
+    val key = str.substring(0, i)
+    val manifestString = str.substring(i + 1)
+    val parser = new ManifestParser
+    val m = parser.parse(parser.manifest, new parser.PackratReader(new CharSequenceReader(manifestString))).getOrElse(
+      sys.error(s"""Invalid manifest string: "$manifestString".""")
+    )
+    Parameter(key)(m)
+  }
```

Comment:
I was hoping too that json serialization can be handled without any custom code. But most scala libraries are typeclass-based so they do not like this `Any` part in maps. I've also checked `genson` and given a try `scala-pickling` but only with `Jackson` I was able to made a solution within assumed time that partially works.

I agree that deserializing `Manifest` is complex and in fact I don't like it :-(
I like your idea with `Pickler[T]`, although one thing that is not clear to me is with case when:

- someone defines generator that needs custom `Pickler`
- (s)he does not want to write custom `Pickler`, but just to use `SerializationPersistor`  

It would throw compilation error (it won't find any suitable `Pickler`), am I right?
So, should I also write stub like:
```
implicit object AnyPickler extends Pickler[Any] {
  def pickle(x: T): Array[Byte]  = sys.error("Pickling for unknown type is disallowed. Please use SerializationPersistor instead.")
  def unpickle(a: Array[Byte]): T
}
```
this one, that will throw error in runtime when trying to use it with json based persistors?
Or just it should up to the user (writing some stub) in this case?

As for `Key`, I think they can be done as they are in this PR. `Manifest` is not persisted - only full path of `Key` is stored, e.g. for `scope` path `dsl.scope` is saved and based only on this `Key` in reconstructed using `Key.parseKey`. 
Or do you think it's a bad idea?
Also, I've chosen `Manifest` and not e.g. `TypeTag` only because:

- there is already conversion function from `Manifest` to Jackson's `JavaType` in `jackson-scala` module
- as far as I know `TypeTag`s are not thread safe in [`Scala 2.10`](https://issues.scala-lang.org/browse/SI-6240?focusedCommentId=71707&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-71707)